src/txn/txn_ckpt.c

```diff
@@ -405,6 +405,24 @@ __checkpoint_reduce_dirty_cache(WT_SESSION_IMPL *session)
 	 */
 	if (cache_size < 10 * WT_MEGABYTE)
 		return;
+
+	/*
+	 * Skip scrubbing if it won't perform at-least some minimum amount of
+	 * work. Scrubbing is supposed to bring down the dirty data to eviction
+	 * checkpoint target before the actual checkpoint starts. Do not perform
+	 * scrubbing if the dirty data to scrub is less than a pre-configured
+	 * size. This size is to an extent based on the configured cache size
+	 * without being too large or too small for large cache sizes. For the
+	 * values chosen, for instance, 100 GB cache will require at-least
+	 * 200 MB of dirty data above eviction checkpoint target, which should
+	 * equate to a scrub phase a few seconds long. Said that, the value of
+	 * 0.2% and 500 MB are still somewhat arbitrary.
+	 */
+	if (__wt_cache_dirty_leaf_inuse(cache) <
+	    ((cache->eviction_checkpoint_target * conn->cache_size) / 100) +
+	    WT_MIN((0.2 * conn->cache_size) / 100, 500 * WT_MEGABYTE))
```

Comment:
This makes more sense now.  I think it would be a bit more readable if it was something like this:
scrub_min = WT_MIN(...);
if (dirty < ((checkpoint+target * cache_size) / 100) + scrub_min)

It makes it easier to digest and see where we may want to tweak it in the future.